include/ozo/transaction.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <ozo/impl/async_start_transaction.h>
+#include <ozo/impl/async_end_transaction.h>
+
+namespace ozo {
+
+template <typename T, typename Query, typename CompletionToken, typename = Require<ConnectionProvider<T>>>
+auto start_transaction(T&& provider, Query&& query, CompletionToken&& token) {
```

Comment:
There are different ways how to start transaction in PostgreSQL.

https://www.postgresql.org/docs/current/static/sql-begin.html
https://www.postgresql.org/docs/current/static/sql-start-transaction.html

Since we support only begin now, why do we restrict users to use another one?

Reply:
Done